### ISSUE TITLE
Sanity check on value encoding?

### ISSUE THREAD

**Issue by joeyhub** [kairosdb/kairosdb#550]
The value encoding for long appears to be based on protocol buffers though I'm not really sure why. Variable length is already supported for the column and it seems to be just using a bit. I assume there was some other intention there (array of values perhaps).

```
	public static void packLong(long value, DataOutput buffer) throws IOException
	{
		// Great trick from http://code.google.com/apis/protocolbuffers/docs/encoding.html#types
		packUnsignedLong((value << 1) ^ (value >> 63), buffer);

	}
```

I might have it wrong but I'm sure the intention here is to just put the signed bit at the end rather than the start which makes sense for a sequence of an arbitrary length.

Bitwise shift operators vary a lot though based on language. In C if the type is unsigned it'll shift everything, if signed it wont shift the first bit (though can be coerced with a cast on what you're shifting to unsigned log log). I always understood in C that triple shifts rotate and I assume they too are type dependent.

In Java double shifts are always unsigned shifts so never touch the signed bit. Tripple shift, which doesn't appear to wrap, is used for unsigned shifts and will shift the signed bit.

Basically run this:

```
    System.out.print(((-2147483648L << 1) ^ (-2147483648L >>> 63)) + "\n");
    System.out.print(((Math.abs(-2147483648L) << 1) ^ (-2147483648L >>> 63)) + "\n");
```

And compare to the output on the page...

**Comment by joeyhub** [comment]
This one is my bad, maybe I have the shift types reversed.
